File: app/src/main/java/com/nikola/jakshic/truesight/data/local/PlayerDao.java
package com.nikola.jakshic.truesight.data.local;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import com.nikola.jakshic.truesight.model.Player;

import java.util.List;

@Dao
public interface PlayerDao {

    @Query("SELECT * FROM player WHERE player.id = :id")
    LiveData<List<Player>> getPlayer(long id);

    @Query("SELECT * FROM player")
    LiveData<List<Player>> getPlayers();

    @Insert
    void insertPlayer(Player player);

    @Delete
    void deletePlayer(Player player);
}

File: app/src/main/java/com/nikola/jakshic/dagger/profile/PlayerDao.kt
package com.nikola.jakshic.dagger.profile

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface PlayerDao {

    @Query("SELECT * FROM players WHERE players.account_id = :id")
    fun getPlayer(id: Long): LiveData<Player>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertPlayer(player: Player)

    @Query("DELETE FROM players WHERE players.account_id = :id")
    fun deletePlayer(id: Long)
}